pkg/R/RDD.R

```diff
@@ -1348,16 +1363,16 @@ setMethod("unionRDD",
           function(x, y) {
             if (x@env$serialized == y@env$serialized) {
               jrdd <- callJMethod(getJRDD(x), "union", getJRDD(y))
-              union.rdd <- RDD(jrdd, x@env$serialized)
+              union.rdd <- RDD(jrdd, x@env$serialized, colNames = x@colNames)
             } else {
               # One of the RDDs is not serialized, we need to serialize it first.
-              if (!x@env$serialized) {
+              if (x@env$serialized == "string") {
```

Comment:
I actually have run that exact test case and the union came out fine. I think this is because PipelinedRDD doesn't draw a distinction between `byte` and `rows` serialization when it's creating the RRDD.

That said, adding this case as an actual unit test is a good idea.  I'll add it now.